src/install.py

```diff
@@ -30,29 +33,43 @@ def update_cache():
     util.write_cache('name', config.DOCKER_CONTAINER_NAME)
 
 
-def main(name=config.DOCKER_CONTAINER_NAME, tag=config.DOCKER_CONTAINER_TAG, tld=config.TOP_LEVEL_DOMAIN):
-    if not util.is_os_supported(OS.FLAVOR):
-        return 1
-
-    if os.path.exists('.cache/INSTALLED'):
-        os.unlink('.cache/INSTALLED')
-
-    # update resolv.conf
+def update_resolvconf():
     if not hasattr(OS, 'DISABLE_MAIN_RESOLVCONF_ROUTINE'):
         if not os.path.exists(RESOLVCONF):
-            open(RESOLVCONF, 'w').write('nameserver 1.1.1.1')
+            content = '# Generated by docker-dns\n'
+            open(RESOLVCONF, 'w').write(content)
 
         dns = docker.NETWORK_GATEWAY
         try:
             dns = OS.DNS
-        except:
+        except Exception:
             # no DNS
             pass
 
-        RESOLVCONF_DATA = open(RESOLVCONF, 'r').read()
-        if '#@docker-dns' not in RESOLVCONF_DATA:
-            RESOLVCONF_DATA = f"options timeout:1 #@docker-dns\nnameserver {dns} #@docker-dns\n{RESOLVCONF_DATA}"
-            open(RESOLVCONF, 'w').write(RESOLVCONF_DATA)
+        name_servers = f'nameserver {dns} #@docker-dns\n' \
+                       f'nameserver 1.1.1.1 #cloudflare\n' \
+                       f'nameserver 8.8.8.8 #google\n'
+        options = 'options timeout:1 #@docker-dns\n'
+        if OS.FLAVOR == 'ubuntu' and config.OS_VERSION >= 18 * 1000:
+            open(RESOLVCONF_HEAD, 'a').write(name_servers)
+            open(RESOLVCONF_TAIL, 'a').write(options)
+            subprocess.run(['resolvconf', '-u'])
```

Comment:
Removed `sudo`, is it ok now?
(still working)

Reply:
The best approach I have found is using `resolvconf`... Couldn't find anything better and scalable for the end user.